File: Guess.cpp
#include <iostream>
#include <string> /* to_string() */
#include <stdlib.h> /* rand(), srand() */
#include <time.h> /* time() */
#include <cmath> /* mathematics */

using namespace std;

/* output text to user */
void println(string output, string vara = "", string varb = "", string varc = "") {
    cout << output << vara << varb << varc << endl;
}

/* take input with label */
string input(string label="", string vara = "", string varb = "", string varc = "") {
    string inp;
    cout << label << vara << varb << varc;
    getline(cin, inp);
    return inp;
}

/* convert string to int */
int parseInt(string num) {
    int _num = std::stoi(num);
    return _num;
}

/* initializing the Game */
class Game {
public:
    int _min = 0; /* Minimum value to guess */
    int _max = 30; /* Maximum value to guess */
    int _chances = 5; /* Total "Try Again" for user */
};

/* How wrong the user guess is?*/
string status(int randm, int guess) {
    Game Guess;

    int distance = guess - randm;
    int factor = Guess._max/5 < 1 ? 1 : round(Guess._max/5);
    string _status;
    if (abs(distance) < factor) {
        _status = distance < 0 ? "Lower" : "Higher";
    } else {
        _status = distance < 0 ? "Much Lower" : "Much Higher";
    }

    return _status;
}


int main() {
    Game Guess;
    println("Welcome to the Guess Game");
    println("Choose a number between: ", to_string(Guess._min), "-", to_string(Guess._max));
    println(to_string(Guess._chances), " chances are available\n");

    int _chance = 1;
    srand(time(NULL));
    int random = rand() % Guess._max + Guess._min;

    /* Game Loop */
    while (_chance <= Guess._chances) {
        int guess = parseInt(input("Chance ", to_string(_chance), ": "));
        if (guess == random) {
            println("\nYou Won!");
            break;
        } else {
            string how_wrong = status(random, guess);
            println("Wrong Guess!, ", how_wrong);
            if (_chance == Guess._chances) println("\nYou Lost!\nActual no: ", to_string(random));
        }
        _chance++;

    }
    return 0;
}

File: Guess.java
import java.io.*;

public class Guess {
    private static final int max = 30;
    private static final int min = 0;
    private static final int maxChance = 5;

    public static void main(String[] args) throws IOException {
        System.out.println("Guess the Number");
        System.out.println("\nGuess the number lying b/w: "+min+"-"+max+" (exclusive)");
        System.out.println("You've got "+maxChance+" chances\n");
        int chance = 1;
        int number = (int)(Math.random()*10);
        while (chance <= maxChance) {
            int guess = Integer.parseInt(input("Chance "+chance+": "));
            if (guess == number) {
                System.out.println("You won");
                break;
            } else {
                String status = findDistance(max, number, guess);
                System.out.println("Guess was: "+status);
            }
            if (chance == maxChance) {
                System.out.println("You lost, number was: "+number);
            }
            chance++;
        }
    }

    public static String findDistance(float _max, int num, int guess) {
        int distance = guess - num;
        int factor = _max/5 < 1 ? 1 : (int) _max/5;
        String status;
        if (Math.abs(distance) < factor) {
            status = distance < 0 ? "Lower" : "Higher";
        } else {
            status = distance < 0 ? "Much Lower" : "Much Higher";
        }
        return status;
    }

    public static String input(String label) throws IOException {
        InputStreamReader read = new InputStreamReader(System.in);
        BufferedReader in = new BufferedReader(read);
        System.out.print(label);
        return in.readLine().trim();
    }
}
